BE/src/main/java/kr/codesquad/issuetracker/service/MilestonesService.java

```diff
@@ -0,0 +1,90 @@
+package kr.codesquad.issuetracker.service;
+
+import static kr.codesquad.issuetracker.exception.ErrorMessage.MILESTONE_NOT_FOUND;
+import static kr.codesquad.issuetracker.exception.ErrorMessage.MILESTONE_NOT_SAVE;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import kr.codesquad.issuetracker.domain.Status;
+import kr.codesquad.issuetracker.domain.issue.Issue;
+import kr.codesquad.issuetracker.domain.label.repository.LabelRepository;
+import kr.codesquad.issuetracker.domain.milestone.Milestone;
+import kr.codesquad.issuetracker.domain.milestone.repository.MilestoneRepository;
+import kr.codesquad.issuetracker.exception.CustomException;
+import kr.codesquad.issuetracker.web.dto.milestone.MilestoneDto;
+import kr.codesquad.issuetracker.web.dto.milestone.MilestoneEditRequestDto;
+import kr.codesquad.issuetracker.web.dto.milestone.MilestoneRequestDto;
+import kr.codesquad.issuetracker.web.dto.milestone.MilestoneResponseDto;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@RequiredArgsConstructor
+@Service
+public class MilestonesService {
+
+	private final MilestoneRepository milestoneRepository;
+
+	private final LabelRepository labelRepository;
+
+	@Transactional(readOnly = true)
+	public MilestoneResponseDto list(String statusValue) {
+		Status status = isStatus(statusValue);
+		return new MilestoneResponseDto(
+			(int) labelRepository.count(),
+			isOpenCount(),
+			isClosedCount(),
+			findMilestonesDtos(status)
+		);
+	}
+
+	private Status isStatus(String status) {
+		String value = status.toUpperCase();
+		if (Status.OPEN.getValue().equals(value)) {
+			return Status.OPEN;
+		}
+		return Status.CLOSED;
+	}
+
+	private int isOpenCount() {
+		return (int) milestoneRepository.countByStatus(Status.OPEN);
+	}
+
+	private int isClosedCount() {
+		return (int) milestoneRepository.countByStatus(Status.CLOSED);
+	}
+
+	private List<MilestoneDto> findMilestonesDtos(Status status) {
```

Comment:
Transactional 어노테이션의 readOnly 옵션 유무에 따른 차이를 알아봅시다.

```suggestion
	@Transactional(readOnly = true)
	private List<MilestoneDto> findMilestonesDtos(Status status) {
```

Reply:
Transactional 어노테이션의 readOnly옵션을 사용하게되면 강제로 플러시하지않는 이상 플러시가 일어나지 않아 등록 수정 삭제등의 db에 반영되는 부분들이 일어나지않고, 읽기 전용이기 때문에 변경감지를 하기위한 스냅샷도 저장하지않는걸로 알고 있습니다!!

- 추가적인 부분 : 스프링 프레임워크가 하이버네이트 세션 플러시 모드를 manual로 설정해서 그렇다고 합니다!!